### PR TITLE
check the return code before on_completed callback

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/client_channel_handler.c [hunk 1]
```diff
@@ -160,10 +136,37 @@ static int s_packet_handler_connack(
             "id=%p: invalid connect return code %d, disconnecting",
             (void *)connection,
             connack.connect_return_code);
-        /* If error code returned, disconnect */
+        /* If error code returned, disconnect, on_completed will be invoked from shutdown process */
```
Comment (bretambrose) [inline source/client_channel_handler.c:139]: Can you link where this is done?

Comment (TingDaoK) [reply source/client_channel_handler.c:139]: https://github.com/awslabs/aws-c-mqtt/blob/master/source/client.c#L181
The return code will not be shown to the client. Only the error code `AWS_ERROR_MQTT_PROTOCOL_ERROR` will be there

Comment (bretambrose) [reply source/client_channel_handler.c:139]: Nice.

**Review by bretambrose — APPROVED** [review]

File: source/client_channel_handler.c [hunk 2]
```diff
@@ -160,10 +136,37 @@ static int s_packet_handler_connack(
...
         aws_channel_shutdown(connection->slot->channel, AWS_ERROR_MQTT_PROTOCOL_ERROR);
```
Comment (graebm) [inline source/client_channel_handler.c:140]: This isn't a protocol error. We should really declare distinct error codes for each of the CONNACK non-zeroes.
AWS_ERROR_MQTT_CONNECTION_REFUSED_UNACCEPTABLE_PROTOCOL_VERSION???

**Pull request merged**